core/src/main/java/hudson/model/Run.java

```diff
@@ -1883,12 +1886,6 @@ protected final void execute(@NonNull RunExecution job) {
                     LOGGER.log(Level.SEVERE, "Failed to save build record", e);
                 }
             }
-
-            try {
-                getParent().logRotate();
-            } catch (Exception e) {
-                LOGGER.log(Level.SEVERE, "Failed to rotate log", e);
-            }
```

Comment:
I moved this into `GlobalBuildDiscarderListener` so that `isLogUpdated() == true` when log rotation runs. It still runs synchronously, it just runs a little bit later now down in `onEndBuilding`. This allows us to check `isLogUpdated()` elsewhere to avoid race conditions with Pipeline builds.